static/js/compose_actions.js

```diff
@@ -98,6 +113,29 @@ function show_box(msg_type, opts) {
     set_focus(msg_type, opts);
 }
 
+let togglable_message_type;
+
+function toggle_message_type(opts) {
+    togglable_message_type = togglable_message_type === "private" ? "stream" : "private";
+    compose_state.set_message_type(togglable_message_type);
+    opts.message_type = togglable_message_type;
+    opts.trigger = "toggle message";
+    opts.stream = compose_state.stream_name();
+    opts.topic = compose_state.topic();
+    opts.private_message_recipient = compose_state.private_message_recipient();
+    reset_message_type_switcher(opts);
+    show_box(togglable_message_type, opts);
+    update_placeholder_text();
+}
```

Comment:
Is there some mechanism through which we'd expect switching the message type to clear any compose box alerts generated by `compose_validate`?

Reply:
Oh boy, I hadn't thought about validation alerts - thanks.

I guess we want to clear and regenerate alerts each time a switch happens?

Reply:
I imagine so; haven't thought about it too much but one at least needs to do the clearing part, plus whatever would happen on-typing (not things that happen on-send); I can't remember but I think there are very few with that model.